src/webgpu/shader/execution/expression/call/builtin/exp.spec.ts

```diff
@@ -27,7 +34,46 @@ T is AbstractFloat, f32, f16, vecN<AbstractFloat>, vecN<f32>, or vecN<f16>
 Returns the natural exponentiation of e1 (e.g. e^e1). Component-wise when T is a vector.
 `
   )
-  .unimplemented();
+  .params(u =>
+    u
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const cmp = (x: number, target: Scalar): Comparator => {
+      return (got, _) => {
+        const c = 3 + 2 * Math.abs(x);
+        const ulpMatch = ulpThreshold(c);
+        const cmp = compare(got, target, ulpMatch);
+        if (cmp.matched) {
+          return cmp;
+        } else {
```

Comment:
nit: don't need the `else` as the `if` returns

Reply:
Done